AuthenticationAndAuthorization/DataAccounts/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DataAccounts.Repositories;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccounts
+{
+    public class UserRepository : IUserRepository
+    {
+        private readonly UserManager<User> _userManager;
+        private readonly SignInManager<User> _signInManager;
+
+        public UserRepository(UserManager<User> userManager,SignInManager<User> signInManager)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+        }
+
+        public async Task Register(User user,string password, string role)
+        {
+            var result = await _userManager.CreateAsync(user,password);
+
+            if (result.Succeeded)
+            {
+                await _userManager.AddToRoleAsync(user, role);
+            }
```

Comment:
Imagine the situation: user manager created a user, then database has outage for 2seconds. when AddToRole happens your db responds failure. Result: you have user without role.

May be we could add here atomicity? For example UnitOfWork pattern or explicit transactions. Please read about atomicity